xmtp_mls/src/storage/sql_key_store.rs

```diff
@@ -94,12 +1051,161 @@ mod tests {
+    //         key_store
+    //             .queue_proposal::<GroupId, ProposalRef, QueuedProposal>(&group_id, &ProposalRef(i), proposal)
+    //             .unwrap();
+    //     }
+
+    //     // Read proposal refs
+    //     let proposal_refs_read: Vec<ProposalRef> = key_store.queued_proposal_refs(&group_id).unwrap();
+    //     assert_eq!(
+    //         (0..10).map(|i| ProposalRef(i)).collect::<Vec<_>>(),
+    //         proposal_refs_read
+    //     );
+
+    //     // Read proposals
+    //     let proposals_read: Vec<(ProposalRef, Proposal)> =
+    //         key_store.queued_proposals(&group_id).unwrap();
+    //     let proposals_expected: Vec<(ProposalRef, Proposal)> = (0..10)
+    //         .map(|i| ProposalRef(i))
+    //         .zip(proposals.clone().into_iter())
+    //         .collect();
+    //     assert_eq!(proposals_expected, proposals_read);
+
+    //     // Remove proposal 5
+    //     key_store.remove_proposal(&group_id, &ProposalRef(5)).unwrap();
+
+    //     let proposal_refs_read: Vec<ProposalRef> = key_store.queued_proposal_refs(&group_id).unwrap();
+    //     let mut expected = (0..10).map(|i| ProposalRef(i)).collect::<Vec<_>>();
+    //     expected.remove(5);
+    //     assert_eq!(expected, proposal_refs_read);
+
+    //     let proposals_read: Vec<(ProposalRef, Proposal)> =
+    //         key_store.queued_proposals(&group_id).unwrap();
+    //     let mut proposals_expected: Vec<(ProposalRef, Proposal)> = (0..10)
+    //         .map(|i| ProposalRef(i))
+    //         .zip(proposals.clone().into_iter())
+    //         .collect();
+    //     proposals_expected.remove(5);
+    //     assert_eq!(proposals_expected, proposals_read);
+
+    //     // Clear all proposals
+    //     key_store
+    //         .clear_proposal_queue::<GroupId, ProposalRef>(&group_id)
+    //         .unwrap();
+    //     let proposal_refs_read: Vec<ProposalRef> = key_store.queued_proposal_refs(&group_id).unwrap();
+    //     assert!(proposal_refs_read.is_empty());
+
+    //     let proposals_read: Vec<(ProposalRef, Proposal)> =
+    //         key_store.queued_proposals(&group_id).unwrap();
+    //     assert!(proposals_read.is_empty());
+
+    //     // assert!(key_store
+    //     //     .own_leaf_nodes::<GroupId, LeafNode>(&group_id)
+    //     //     .unwrap()
+    //     //     .is_empty());
+
+    //     // key_store
+    //     //     .append_own_leaf_node::<GroupId, LeafNode>(&group_id, &new_key_package.leaf_node())
+    //     //     .unwrap();
+
+    //     // assert!(!key_store
+    //     //     .own_leaf_nodes::<GroupId, LeafNode>(&group_id)
+    //     //     .unwrap()
+    //     //     .is_empty());
+
+    //     // key_store
+    //     //     .clear_own_leaf_nodes::<GroupId>(&group_id)
+    //     //     .unwrap();
+
+    //     // assert!(key_store
+    //     //     .own_leaf_nodes::<GroupId, LeafNode>(&group_id)
+    //     //     .unwrap()
+    //     //     .is_empty());
+    // }
```

Comment:
@franziskuskiefer I can't seem to get the aad or the proposal ref tests working. What am I missing here? Mind helping me out?